src/net.cpp

```diff
@@ -2865,3 +2867,25 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(CNode* node, std::string& command, std::vector<unsigned char>& data, bool is_incoming)
+{
+    int64_t time = GetTimeMicros();
+
+    fs::path base_path = GetDataDir() / "peer_logging" / node->addr.ToString();
+    boost::filesystem::create_directories(base_path);
```

Comment:
Does this mean we try to create the directory every time we send/receive a message? Is that a problem?

Reply:
This is ok.  As per the boost docs: "Creation failure because [base_path] resolves to an existing directory shall not be treated as an error."

Reply:
I can't imagine this is a performance hit, but if anyone is worried I guess this could be moved to the CNode's constructor.  I would want to first be really sure that a node's address won't ever change.